Makefile

```diff
@@ -45,6 +45,12 @@ linux-prepare::
 	if [ "x$$BINFMT_WARN" = "xyes" ]; then \
 		cat Documentation/binfmt_misc-warning-Linux.txt ; \
 	fi; \
+	if  [ -f build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO).sh ] || [ -f build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO)-$(LINUX_DISTRO_RELEASE).sh ]  &&  [ "$(NO_SUDO)" = "false" ]; then \
```

Comment:
Please see the [**test**(1) man page](https://linux.die.net/man/1/test): `test` (aka `[`) supports a `-o` option for "or", which avoids the need to use `[ ... ] || [ ... ]`:

    if [ \( -f build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO).sh -o -f build-tools/scripts/dependencies/linux-prepare-$(LINUX_DISTRO)-$(LINUX_DISTRO_RELEASE).sh \) -a "$(NO_SUDO)" = "false" ] ; then

Note that `(` is escaped as `\(`, as bash doesn't like it otherwise:

    $ FOO=bar; if [ ( -d bin -o -f bin/Version ) -a "$FOO" = "bar" ]; then echo yay; fi
    -bash: syntax error near unexpected token `-d'

I have no idea if additional backspacing would be needed within make context. That question may be moot.

I also don't understand why this block is added. The scripts themselves should check for `NO_SUDO`, as it's the scripts themselves which can print out meaningful information such as *which packages to install*. That can't be done here.